src/cli.ts

```diff
@@ -9,27 +9,72 @@ import * as prompt from './prompt'
 import { JsonSchemaForTheTypeScriptCompilersConfigurationFile } from './tsconfig-schema'
 import { JsonSchemaForTheTsLintConfigurationFiles } from './tslint-schema'
 
-async function main(): Promise<void> {
-    console.log(['', 'Welcome to the Sourcegraph extension creator!', ''].join('\n'))
+interface Repository {
+    type: string
+    url: string
+    directory?: string
+}
 
-    if (!(await exists('.git'))) {
-        console.log('📘 .git directory not found, initilizing git repository')
-        await exec('git', ['init'])
+async function main(): Promise<void> {
+    async function getHTTPSGitRemoteURL(): Promise<URL | undefined> {
+        try {
+            return new URL(
+                // Simpler than using URL regex
+                exec
+                    .shellSync('git remote -v')
+                    .stdout.split('\n')[0]
+                    .split('\t')[1]
+                    .split(' ')[0]
+                    .replace(':', '/')
+                    .replace('git@', 'https://')
+                    .replace('https//', 'https:/')
+            )
+        } catch (e) {
+            return undefined
+        }
     }
 
+    console.log(['', 'Welcome to the Sourcegraph extension creator!', ''].join('\n'))
+
     let name: string | undefined
+    let repository: Repository | undefined
     let title: string | undefined
     let description: string | undefined
     let publisher: string | undefined
+    let license: string | undefined
 
     try {
-        ;({ name, title, description, publisher } = JSON.parse(await readFile('package.json', 'utf-8')))
+        ;({ name, title, description, publisher, repository, license } = JSON.parse(
+            await readFile('package.json', 'utf-8')
+        ))
     } catch (err) {
         if (err.code !== 'ENOENT') {
             throw err
         }
     }
 
+    if (!(await exists('.git'))) {
+        console.log('📘 .git directory not found, initializing git repository.\n')
+        await exec('git', ['init'])
+    }
+
+    if (repository) {
+        console.log(`Extension ${repository.type} repository url is "${repository.url}"`)
+    } else {
+        const url = await getHTTPSGitRemoteURL()
+
+        if (url) {
+            repository = {
+                type: 'git',
+                url: url!.href,
+            }
+        } else {
+            console.log(
+                '📘  Unable to set the "repository" field in package.json as a git remote was not found. You should set this manually before publishing your extension.\n'
```

Comment:
As a user reading this, I might think the `repository` field is mandatory. How about:

> Not setting the "repository" field in package.json since this looks like a new repository. You can add it later anytime.

Reply:
I don't think it's bad if users think that. If a user does not set it, we are unable to get global references from their repo, potentially breaking their use of the API